### PR TITLE
KFSPTS-22886 create concur admin screen to get a new request and refresh token

### PR DESCRIPTION
**Pull request by jhulslander**
Please note there is a related non prod SQL PR

### CODE REVIEW

**Comment by ajd299** [comment]
Please see my comments. Also, what is the expected behavior if Concur is down?

**Review by ajd299 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/cornell/kfs/concur/web/struts/ConcurManageRefreshTokenAction.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        ConfigurationService configService = getConfigurationService();
+
+        concurTokenForm.setNonProdWarning(configService
+                .getPropertyValueAsString(ConcurKeyConstants.MESSAGE_CONCUR_REFRESH_TOKEN_NONPROD_WARNING));
+        concurTokenForm.setDisplayNonProdWarning(!isProduction());
+
+        String refreshTokenUpdateDate = MessageFormat.format(
+                configService.getPropertyValueAsString(ConcurKeyConstants.MESSAGE_CONCUR_TOKEN_DATE), "refresh",
+                getTokenUpdateTimestamp(ConcurConstants.ConcurOAuth2.WebServiceCredentialKeys.REFRESH_TOKEN).toString());
+        String updateRefeshSuccessMessage = MessageFormat.format(
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/concur/web/struts/ConcurManageRefreshTokenAction.java:97]: ok

File: src/main/java/edu/cornell/kfs/concur/web/struts/ConcurManageRefreshTokenAction.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                configService.getPropertyValueAsString(ConcurKeyConstants.MESSAGE_CONCUR_TOKEN_UPDATE_SUCCESS),
+                "refresh");
+        concurTokenForm.setRefreshTokenUpdateDate(refreshTokenUpdateDate);
+        concurTokenForm.setUpdateRefreshTokenMessage(updateRefeshSuccessMessage);
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/concur/web/struts/ConcurManageRefreshTokenAction.java:101]: ok

File: src/main/resources/edu/cornell/kfs/concur/cu-spring-concur.xml [hunk 3]
```diff
@@ -18,7 +18,12 @@
 	<bean id="concurModuleConfiguration-parentBean" class="org.kuali.kfs.sys.FinancialSystemModuleConfiguration"
 		abstract="true">
 		<property name="namespaceCode" value="KFS-CNCR" />
-		<property name="initializeDataDictionary" value="false" />
+		<property name="initializeDataDictionary" value="true" />
+		<property name="dataDictionaryPackages">
+      <list merge="true">
+                <value>classpath:edu/cornell/kfs/concur/web/struts/form/datadictionary/*.xml</value>
+      </list>
+    </property>
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/resources/edu/cornell/kfs/concur/cu-spring-concur.xml:26]: ok

File: src/main/resources/edu/cornell/kfs/concur/web/struts/form/datadictionary/ConcurManageRefreshTokenForm.xml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns="http://www.springframework.org/schema/beans"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:p="http://www.springframework.org/schema/p"
+	xsi:schemaLocation="http://www.springframework.org/schema/beans         http://www.springframework.org/schema/beans/spring-beans-2.0.xsd">
+
+
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/resources/edu/cornell/kfs/concur/web/struts/form/datadictionary/ConcurManageRefreshTokenForm.xml:7]: ok

File: src/main/resources/edu/cornell/kfs/concur/web/struts/form/datadictionary/ConcurManageRefreshTokenForm.xml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+				<property name="mask" value="false" />
+				<property name="maskFormatter">
+					<bean parent="MaskFormatterLiteral"
+						p:literal="*****************" />
+				</property>
+			</bean>
+		</property>
+	</bean>
+         
+
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/resources/edu/cornell/kfs/concur/web/struts/form/datadictionary/ConcurManageRefreshTokenForm.xml:37]: ok

File: src/main/webapp/jsp/concur/manageRefreshToken.jsp [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+					<div style="font-weight: bold">
+						<c:out value="${KualiForm.nonProdWarning}" />
+					</div>
+				</div>
+			</div>
+		</c:when>
+	</c:choose>
+
+	<div class="main-panel">
+		<div class="center" style="margin: 30px 0;" ]>
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/webapp/jsp/concur/manageRefreshToken.jsp:24]: no it shouldn't be there, I am surprised it didn't cause an error, I will remove

File: src/main/webapp/jsp/concur/manageRefreshToken.jsp [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+	docTitle="Concur OAuth2 Refresh and Request Token Management"
+	transactionalDocument="false"
+	htmlFormAction="concur/manageRefreshToken" errorKey="*">
+	
+	<c:set var="refreshTokenFormAttributes" value="${DataDictionary.ConcurManageRefreshTokenForm.attributes}"/>
+
+	<c:choose>
+		<c:when test="${KualiForm.displayNonProdWarning}">
+			<div class="main-panel">
+				<div class="center" style="margin: 30px 0;" ]>
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/webapp/jsp/concur/manageRefreshToken.jsp:14]: will remove

**Comment by jhulslander** [comment]
I've made a number of changes based on the feedback.  I've also adjusted the log level to debug on action class logging.  This is ready for review again

**Review by ajd299 — CHANGES_REQUESTED** [review]
I found one minor format issue. After that I think it is ready to be merged

**Comment by ajd299** [comment]
Looks good will merge when unit tests complete

**Comment by jhulslander** [comment]
I removed the square bracket, this is ready for review again

**Review by ajd299 — APPROVED** [review]

**Pull request merged**